File: src/main/java/com/arcao/geocaching4locus/util/feedback/collector/ConstantsCollector.java
package com.arcao.geocaching4locus.util.feedback.collector;

import java.lang.reflect.Field;

public class ConstantsCollector extends Collector {
	private final String prefix;
	private final Class<?> source;

	public ConstantsCollector(Class<?> source, String prefix) {
		this.prefix = prefix;
		this.source = source;
	}

	@Override
	public String getName() {
		return prefix + " CONSTANTS";
	}

	@Override
	public String collect() {
		final StringBuilder result = new StringBuilder();

		final Field[] fields = source.getFields();
		for (final Field field : fields) {
			result.append(field.getName()).append("=");
			try {
				result.append(field.get(null).toString());
			} catch (IllegalArgumentException | IllegalAccessException e) {
				result.append("N/A");
			}
			result.append("\n");
		}

		return result.toString();
	}
}

File: app/src/main/java/com/arcao/feedback/collector/ConstantsCollector.kt
package com.arcao.feedback.collector

import java.util.Arrays

class ConstantsCollector(private val source: Class<*>, private val prefix: String) : Collector() {

    override val name: String
        get() = "$prefix CONSTANTS"

    override suspend fun collect(): String {
        val result = StringBuilder()

        source.fields.forEach { field ->
            result.append(field.name).append("=")
            try {
                result.append(
                    when (val value = field.get(null)) {
                        is Array<*> -> Arrays.toString(value)
                        else -> value.toString()
                    }
                )
            } catch (e: IllegalArgumentException) {
                result.append("N/A")
            } catch (e: IllegalAccessException) {
                result.append("N/A")
            }
            result.append("\n")
        }

        return result.toString()
    }
}
